source/globals.d.ts

```diff
@@ -1,14 +1,57 @@
-declare const browser: AnyObject;
+type ArgumentTypes<TFunction> = TFunction extends (...args: infer TInferredArgs) => infer TInferredReturnType ? TInferredArgs : never;
 
-type AnyObject = {[key: string]: any};
+// Returns the rest args of a function minus the first argument.
+type TailArgs<TFunction extends any[]> = ((...args: TFunction) => any) extends ((
+	_: infer First,
+	...rest: infer Rest
+) => any)
+	? TFunction extends any[] ? Rest : ReadonlyArray<Rest[number]>
+	: [];
+
+// TODO: Type anything that is of type AnyObject
+// for browser, we can cherry pick from the chrome namespace from @types/chrome
+// ensuring that functions return Promises where applicable.
+type AnyObject = Record<string, any>;
 
 declare module 'select-dom' {
 	import 'select-dom';
 
-	function select(selector: string, parent?: any): any | null;
+	function select<T extends HTMLElement = HTMLElement>(selector: string, parent?: string|Element|DocumentFragment): T | null;
 	namespace select {
 		export function exists(selector: string, parent?: Element): boolean;
-		export function all(selector: string, parent?: Element | Element[] | NodeList | DocumentFragment): any[];
+		export function all<T extends HTMLElement = HTMLElement>(selector: string, parent?: Element | Element[] | NodeList | DocumentFragment): T[];
 	}
 	export = select;
 }
+
+declare module 'webext-options-sync' {
+	export interface DefineOptions {
```

Comment:
Does this need to be exported? Also it’d be great to see this moved into the module and out of Refined GitHub

Reply:
There does not appear to be any `@types` defined for it, so I created them. Actually I just saw that it's one of your packages 😃 Same goes for `webext-domain-permission-toggle` and `webext-dynamic-content-scripts`

Reply:
I'll pull it out and I can PR to your modules. Will get to that tomorrow probably.